### ISSUE TITLE
custom kernel possibly messing with OpenVPN

### ISSUE THREAD

**Issue by trandrew** [jaketame/plexacd#9]
So I just tested this by doing a fresh install of Ubuntu 16.04 LTS. After the fresh install, I used this guide http://www.htpcguides.com/force-torrent-traffic-vpn-split-tunnel-debian-8-ubuntu-16-04/ to set up split tunnel VPN. At the end of the tutorial, everything checks out and "sudo -u vpn -i -- curl ipinfo.io" pulls up the right info. After this I installed rclone latest beta, then your scripts. After installing the custom kernel, I rebooted and uname -a verified that it loaded the custom 4.4.52. However, the previous command to check the vpn user's curl info timed out. So to check if the custom kernel was the issue, I rebooted into the original generic kernel and the curl command worked again. I know you've said the custom kernel is mostly the defaults with a couple adjustments, but is there any reason why it would conflict with my split tunneling?

**Comment by jaketame** [comment]
I suspect the package ipinfo.io is missing, net filter is also missing. Looking to get this added shortly.

**Comment by jaketame** [comment]
hmm, I ran this command using the same kernel with no issues.

Do you get any errors on /var/log/messages?

**Comment by trandrew** [comment]
I don't seem to have /var/log/messages in my logs folder, but /var/log/syslog doesn't have anything that would help either.

```
plex@TreyFive:~$ uname -a
Linux TreyFive 4.4.52 #1 SMP Fri Mar 3 15:38:10 GMT 2017 x86_64 x86_64 x86_64 GNU/Linux
plex@TreyFive:~$ sudo -u vpn -i -- curl ipinfo.io
[sudo] password for plex: 
curl: (6) Could not resolve host: ipinfo.io
plex@TreyFive:~$ curl ipinfo.io
{
  "ip": "7x.xxx.xx.x4",
  "hostname": "x.com",
  "city": "North Hollywood",
  "region": "California",
  "country": "US",
  "loc": "x,x",
  "org": "AS20001 Time Warner Cable Internet LLC",
  "postal": "x"
```

**Comment by jaketame** [comment]
Looks like VPN user is having issues.

Su - vpn 
Then try the curl as the user.

On 9 Mar 2017, at 17:43, Andrew Tran <notifications@github.com> wrote:

I don't seem to have /var/log/messages in my logs folder, but /var/log/syslog doesn't have anything that would help either.

plex@TreyFive:~$ uname -a
Linux TreyFive 4.4.52 #1 SMP Fri Mar 3 15:38:10 GMT 2017 x86_64 x86_64 x86_64 GNU/Linux
plex@TreyFive:~$ sudo -u vpn -i -- curl ipinfo.io
[sudo] password for plex: 
curl: (6) Could not resolve host: ipinfo.io
plex@TreyFive:~$ curl ipinfo.io
{
  "ip": "7x.xxx.xx.x4",
  "hostname": "x.com",
  "city": "North Hollywood",
  "region": "California",
  "country": "US",
  "loc": "x,x",
  "org": "AS20001 Time Warner Cable Internet LLC",
  "postal": "x"

**Comment by trandrew** [comment]
Nope getting the same "Could not resolve host" issue. Just tried creating another user to run deluge behind the split tunnel VPN and it gets the timeout error.

**Comment by jaketame** [comment]
Can you ping any hosts when your another user?

On 9 Mar 2017, at 18:03, Andrew Tran <notifications@github.com> wrote:

Nope getting the same "Could not resolve host" issue. Just tried creating another user to run deluge behind the split tunnel VPN and it gets the timeout error.

**Comment by trandrew** [comment]
Created new user (non-VPN split tunnel) and everything works.

```
plex@TreyFive:~$ sudo useradd newuser
[sudo] password for plex: 
plex@TreyFive:~$ su newuser
Password: 
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

newuser@TreyFive:/home/plex$ curl ipinfo.io
{
  "ip": "7x.xxx.xxx.x4",
  "hostname": "x.com",
  "city": "North Hollywood",
  "region": "California",
  "country": "US",
  "loc": "x,x",
  "org": "AS20001 Time Warner Cable Internet LLC",
  "postal": "x"
}newuser@TreyFive:/home/plex$ ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=57 time=9.13 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=57 time=9.66 ms
^C
--- 8.8.8.8 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1001ms
rtt min/avg/max/mdev = 9.133/9.399/9.665/0.266 ms
```

In the guide I linked in the original post, under "Change Reverse Path Filtering" it mentions the kernel routing the traffic correctly. Could the issue be here? For the record, I haven't applied your sysctl.conf file in the scripts.

**Comment by jaketame** [comment]
I will read the guide properly and refer back to you.

On 9 Mar 2017, at 18:16, Andrew Tran <notifications@github.com> wrote:

Created new user (non-VPN split tunnel) and everything works.

plex@TreyFive:~$ sudo useradd newuser
[sudo] password for plex: 
plex@TreyFive:~$ su newuser
Password: 
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

newuser@TreyFive:/home/plex$ curl ipinfo.io
{
  "ip": "7x.xxx.xxx.x4",
  "hostname": "x.com",
  "city": "North Hollywood",
  "region": "California",
  "country": "US",
  "loc": "x,x",
  "org": "AS20001 Time Warner Cable Internet LLC",
  "postal": "x"
}newuser@TreyFive:/home/plex$ ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=57 time=9.13 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=57 time=9.66 ms
^C
--- 8.8.8.8 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1001ms
rtt min/avg/max/mdev = 9.133/9.399/9.665/0.266 ms
In the guide I linked in the original post, under "Change Reverse Path Filtering" it mentions the kernel routing the traffic correctly. Could the issue be here? For the record, I haven't applied your sysctl.conf file in the scripts.

**Comment by jaketame** [comment]
This is resolved in the new kernel 4.4.61.2 https://github.com/jaketame/plexgsuite/tree/master/kernel
